internal/services/healthcare/healthcare_dicom_resource.go

```diff
@@ -373,3 +532,103 @@ func flattenDicomServicePrivateEndpoint(input *[]dicomservices.PrivateEndpointCo
 	}
 	return results
 }
+
+func expandStorageConfiguration(input []interface{}) *dicomservices.StorageConfiguration {
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+	storageSettings := input[0].(map[string]interface{})
+
+	var storageResourceId *string
+	if v, ok := storageSettings["storage_account_id"]; ok {
+		storageResourceId = pointer.To(v.(string))
+	}
+
+	var fileSystemName *string
+	if v, ok := storageSettings["file_system_name"]; ok {
+		fileSystemName = pointer.To(v.(string))
+	}
+
+	return &dicomservices.StorageConfiguration{
+		FileSystemName:    fileSystemName,
+		StorageResourceId: storageResourceId,
+	}
+}
+
+func flattenStorageConfiguration(configuration *dicomservices.StorageConfiguration) interface{} {
+	if configuration == nil {
+		return []interface{}{}
+	}
+
+	result := make(map[string]interface{})
+	if configuration.FileSystemName != nil {
+		result["file_system_name"] = *configuration.FileSystemName
+	}
```

Comment:
We could use `pointer.From` here for the nil check 
```suggestion
	result["file_system_name"] = pointer.From(configuration.FileSystemName)
```

Reply:
Thanks, updated, the flatten function (used by the resource & data source)